src/library_pthread.js

```diff
@@ -416,6 +422,8 @@ var LibraryPThread = {
     Atomics.store(HEAPU32, (pthread.threadInfoStruct + {{{ C_STRUCTS.pthread.attr }}} + 20) >> 2, threadParams.schedPolicy);
     Atomics.store(HEAPU32, (pthread.threadInfoStruct + {{{ C_STRUCTS.pthread.attr }}} + 24) >> 2, threadParams.schedPrio);
 
+    Atomics.store(HEAPU32, (pthread.threadInfoStruct + {{{ C_STRUCTS.pthread.locale }}}) >> 2, __global_locale_ptr);
```

Comment:
I like it, implemented the non-sketch version in https://github.com/kripken/emscripten/pull/4813/commits/6cea7ccd64f07e5c2d2568113de429443c248733

Didn't remove the constructor because 1) the pthread stub case still needs it, and 2) there's no good place to instantiate the main thread this way. I still think this is an improvement though.